mssql-ballerina/Client.bal

```diff
@@ -0,0 +1,227 @@
+#                       were provided when a user signed in to the computer or network.
+# + database - System Identifier or the Service Name of the database
+# + user - Name of a user of the database
+# + password - Password for the user
+# + options - Mssql database specific JDBC options
+# + connectionPool - The `sql:ConnectionPool` object to be used within 
+#         the jdbc client. If there is no connectionPool provided, 
+#         the global connection pool will be used
+type ClientConfiguration record {|
+    string host;
+    string? instanceName;
+    int port;
+    boolean? integratedSecurity;
+    string? user;
+    string? password;
+    string? database;
+    Options? options;
+    sql:ConnectionPool?  connectionPool;
+|};
+
+# MsSQL database options.
+#
+# + ssl - SSL Configuration to be used
+# + useXADatasource - Boolean value to enable XADatasource
+# + socketTimeoutInSeconds - The number of milliseconds to wait before a timeout is occurred 
+#                   on a socket read or accept. The default value is 0, which means 
+#                   infinite timeout.
+# + queryTimeoutInSeconds - The number of seconds to wait before a timeout has occurred on a 
+#                  query. The default value is -1, which means infinite timeout. 
+#                  Setting this to 0 also implies to wait indefinitely.
+# + loginTimeoutInSeconds - The number of seconds the driver should wait before timing out a 
+#                 failed connection. A zero value indicates that the timeout is the 
+#                 default system timeout, which is specified as 15 seconds by default. 
+#                 A non-zero value is the number of seconds the driver should wait 
+#                 before timing out a failed connection.
+
+public type Options record {|
+    SSLConfig ssl = {};
+    boolean useXADatasource = false;
+    int socketTimeoutInSeconds?;
+    int queryTimeoutInSeconds?;
+    int loginTimeoutInSeconds?;
+|};
+
+# SSL Configuration to be used when connecting to Mssql server.
+#
+# + clientCertKeystore - Keystore configuration of the client certificates
+# + trustCertKeystore - Keystore configuration of the trust certificates
+# + encrypt - encryption for all the data sent between the client and the server if the server has a certificate installed
+# + trustServerCertificate - If "true", the SQL Server SSL certificate is automatically trusted when the communication layer is encrypted using TLS.
+# + trustStore - The path (including filename) to the certificate trustStore file. The trustStore file contains the list of certificates that the client trusts.
+# + trustStorePassword - The password used to check the integrity of the trustStore data.
+ 
+public type SSLConfig record {|
+    boolean encrypt?;
+    boolean trustServerCertificate?;
+    string trustStore?;
+    string trustStorePassword?;
+    crypto:KeyStore clientCertKeystore?;
+    crypto:KeyStore trustCertKeystore?;
```

Comment:
Shall we stick with the https://github.com/ballerina-platform/module-ballerina-grpc/blob/master/grpc-ballerina/client_endpoint.bal#L235-L253 format for the `SecureSocket` configurations

Reply:
We may not need all params

Reply:
Do you able to update this as we discussed earlier